### DISCUSSION TITLE
Long delay before querying table with many unique relations

### DISCUSSION THREAD

**Discussion by mattkindy** [prisma/prisma#19103] [Q&A] [answered]
Hey all -- recently changed a large number of relations between two models from 1-to-many to 1-to-1 with unique keys. You can see the schema in some detail here:

## QuoteRequest
This is just a subset of the model to give a sense of the setup.

### Relations
<img width="1342" alt="image" src="https://user-images.githubusercontent.com/35696388/236581666-cc124b8d-9418-4c03-a91a-f20f324825ec.png">

### Foreign keys
<img width="777" alt="image" src="https://user-images.githubusercontent.com/35696388/236581792-cf2b5f9b-336c-4771-b96d-8d3cc09d7521.png">

## RangeSpecMinMax
<img width="1289" alt="image" src="https://user-images.githubusercontent.com/35696388/236581722-332a57c2-e4eb-43f3-a5e5-4be367a1c2b6.png">

Since then, when we query this table (with all the unique foreign keys) with the prisma client

```bash
2023-05-05T17:05:49.460-05:00	[34mdebug[39m: {
2023-05-05T17:05:49.460-05:00	"query": "...",
2023-05-05T17:05:49.460-05:00	"params": "[\"ckzpxf3g7066w0704yxf4bzsr\"]",
2023-05-05T17:05:49.460-05:00	"duration": "4 ms"
2023-05-05T17:05:49.460-05:00	}

2023-05-05T17:05:53.900-05:00	[34mdebug[39m: Flushing 2 events

2023-05-05T17:05:55.927-05:00	[34mdebug[39m: {
2023-05-05T17:05:55.927-05:00	"query": "SELECT ... WHERE (... = $1 AND ... IN ($2)) OFFSET $3",
2023-05-05T17:05:55.927-05:00	"params": "[\"NA\",\"ckzpxf3g7066w0704yxf4bzsr\",0]",
2023-05-05T17:05:55.927-05:00	"duration": "20 ms"
2023-05-05T17:05:55.927-05:00	}
```

I see a significant amount of empty time between the end of the previous query and the beginning of the next. The only clue I have is this "Flushing [N] events" (nearly always N=2), which doesn't appear anywhere that I can find in the Prisma source code. It doesn't affect any other table, and the SQL statement when run directly seems reasonably performant. Any ideas?

### Additional Details
- **Schema**: 142 models, many relations
- **Prisma client version**: 4.13
- **Preview features**: Using JSON Protocol
- **Execution environment**: AWS Lambda, NodeJS 16.
- **Build**: Webpack
- **Database connection configuration**: 
  - Using pgbouncer for connection pooling to Postgres RDS. 
  - In this snapshot 5 connections for the client used in this lambda

**Comment by mattkindy** [comment]
From doing a little digging, it seems like whatever is happening is occurring at the engine level, since I don't see any delays outside of the delegated query promise.

**Reply by mattkindy** [reply]
I do also see increased peak memory usage (delta of 3GB+) in this scenario as well

**Reply by nurul3101** [reply]
Do you mind creating a [Bug Report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) for this so that our engineering team can have a look?

**Reply by nurul3101** [reply, marked as answer]
Thanks for creating the Bug Report
For reference here is the GitHub Issue: https://github.com/prisma/prisma/issues/19160
